dpnp/dpnp_iface_logic.py

```diff
@@ -572,48 +546,109 @@ def any(a, /, axis=None, out=None, keepdims=False, *, where=True):
+    rtol : {dpnp.ndarray, usm_ndarray, scalar}, optional
+        The relative tolerance parameter. Default: ``1e-05``.
+    atol : {dpnp.ndarray, usm_ndarray, scalar}, optional
+        The absolute tolerance parameter. Default: ``1e-08``.
+    equal_nan : bool
+        Whether to compare ``NaNs`` as equal. If ``True``, ``NaNs`` in `a` will
+        be considered equal to ``NaNs`` in `b` in the output array.
+
+    Returns
+    -------
+    out : dpnp.ndarray
+        Returns a boolean array of where `a` and `b` are equal within the given
+        tolerance.
 
     See Also
     --------
-    :obj:`dpnp.allclose` : Returns True if two arrays are element-wise equal
-                           within a tolerance.
+    :obj:`dpnp.allclose` : Returns ``True`` if two arrays are element-wise
+                           equal within a tolerance.
 
     Examples
     --------
     >>> import dpnp as np
-    >>> x1 = np.array([1e10,1e-7])
-    >>> x2 = np.array([1.00001e10,1e-8])
-    >>> out = np.isclose(x1, x2)
-    >>> [i for i in out]
-    [True, False]
+    >>> a = np.array([1e10, 1e-7])
+    >>> b = np.array([1.00001e10, 1e-8)
+    >>> np.isclose(a, b)
+    array([ True, False])
 
-    """
+    >>> a = np.array([1e10, 1e-8])
+    >>> b = np.array([1.00001e10, 1e-9])
+    >>> np.isclose(a, b)
+    array([ True,  True])
 
-    # x1_desc = dpnp.get_dpnp_descriptor(x1)
-    # x2_desc = dpnp.get_dpnp_descriptor(x2)
-    # if x1_desc and x2_desc:
-    #     result_obj = dpnp_isclose(
-    #         x1_desc, x2_desc, rtol, atol, equal_nan
-    #     ).get_pyobj()
-    #     return result_obj
+    >>> a = np.array([1e10, 1e-8])
+    >>> b = np.array([1.0001e10, 1e-9])
+    >>> np.isclose(a, b)
+    array([False,  True])
 
-    return call_origin(
-        numpy.isclose, x1, x2, rtol=rtol, atol=atol, equal_nan=equal_nan
+    >>> a = np.array([1.0, np.nan])
+    >>> b = np.array([1.0, np.nan])
+    >>> np.isclose(a, b)
+    array([ True, False])
+    >>> np.isclose(a, b, equal_nan=True)
+    array([ True,  True])
+
+    >>> a = np.array([0.0, 0.0])
+    >>> b = np.array([1e-8, 1e-7])
+    >>> np.isclose(a, b)
+    array([True, False])
+    >>> b = np.array([1e-100, 1e-7])
+    >>> np.isclose(a, b, atol=0.0)
+    array([False, False])
+
+    >>> a = np.array([1e-10, 1e-10])
+    >>> b = np.array([1e-20, 0.0])
+    >>> np.isclose(a, b)
+    array([ True,  True])
+    >>> b = np.array([1e-20, 0.999999e-10])
+    >>> np.isclose(a, b atol=0.0)
+    array([False,  True])
+
+    """
+
+    dpnp.check_supported_arrays_type(a, b, scalar_type=True)
+    dpnp.check_supported_arrays_type(
+        rtol, atol, scalar_type=True, all_scalars=True
     )
 
+    if dpnp.isscalar(b):
+        dt = dpnp.result_type(
+            b, dpnp.default_float_type(sycl_queue=a.sycl_queue)
+        )
+        b = dpnp.asarray(
+            b, dtype=dt, sycl_queue=a.sycl_queue, usm_type=a.usm_type
+        )
+    elif b.dtype.kind == "i":
+        dt = dpnp.result_type(
```

Comment:
Will it always result in `dpnp.default_float_type` or thee is the use case when the result differs from  `dpnp.default_float_type` ?

Reply:
Changed to weak float dtype